Crackmes/Write-ups/OMTG-Android-App/writeup_Dynamic_instrumentation_with_frida_gadget.md

```diff
@@ -0,0 +1,215 @@
+# Dynamic instrumentation with frida gadget
+
+> Author: Jon-Anthoney de Boer
+
+## Why use the frida gadget?
+
+It can be useful to be able to make use of a non-rooted test device when carrying out security testing of an android application. This can be achieved by injecting the frida gadget into a target application to provide support for dynamic binary instrumentation. The steps required to instrument the target application are a little more involved when making use of the frida gadget on a non-rooted device than they are where the tester has recourse to a rooted test device and so can make use of the frida-server component.
```

Comment:
There is a connection between a non-rooted device and using Frida gadget, but that isn't explained here. Of course, on a rooted device you would use Frida server, and that's not possible on a non-rooted device. Maybe it would be good to explain that here, if it isn't explained above.

Not everybody may know what "the frida gadget" is. This could use a little introduction.

Below you describe how to patch an apk to contain the Frida gadget. I never tried it, but it seems [objection can do that with a single command](https://github.com/sensepost/objection/wiki/Patching-Android-Applications#patching---patching-an-apk). That may be good to mention somewhere.

Reply:
Hey @Sjord - thanks for your review and feedback. Updates provided, please let me know if these are satisfactory. Cheers

Reply:
Hey @Sjord - *ping* =)

Reply:
Yes, excellent introduction. It provides some context before diving in about frida gadget.